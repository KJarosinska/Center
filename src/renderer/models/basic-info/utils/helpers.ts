import { SimCard } from "Renderer/models/basic-info/interfaces"

export const getActiveNetworkFromSim = (simCards: SimCard[] = []) => {
  const activeSimCard = simCards.filter(simCard => simCard.active)
  if (activeSimCard.length === 0) {
    return "No connection"
  }
  return activeSimCard[0].network
}
