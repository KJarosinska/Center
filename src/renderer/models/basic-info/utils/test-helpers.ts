import { IpcRequest } from "Common/requests/ipc-request.enum"

export const commonCalls = {
  [IpcRequest.GetDeviceInfo]: Promise.resolve({
    name: "Ziemniaczek",
    modelName: "U12300000",
    modelNumber: "A1239999",
    serilaNumber: "a-b-3d",
    osVersion: "0.123v",
    osUpdateDate: "12-12-2003",
  }),
  [IpcRequest.GetNetworkInfo]: Promise.resolve({
    simCards: [
      {
        active: true,
        carrier: "AAAAAAAAAAAA",
        iccid: 1234,
        imei: 5678,
        meid: 8765,
        network: "Y-Mobile",
        networkLevel: 0.5,
        number: 12345678,
        seid: "1234",
        slot: 1,
      },
      {
        active: false,
        carrier: "BBBBBBBBBBBB",
        iccid: 412,
        imei: 42,
        meid: 1410,
        network: "X-Mobile",
        networkLevel: 0.69,
        number: 7001234523,
        seid: "x123",
        slot: 2,
      },
    ],
  }),
  [IpcRequest.GetStorageInfo]: Promise.resolve({
    capacity: 9001,
    available: 99999999999999,
    categories: [
      { label: "music", filesCount: 1233333, size: 999999999 },
      { label: "storage", filesCount: 100000, size: 999999999 },
    ],
  }),
  [IpcRequest.GetBatteryInfo]: Promise.resolve({
    level: 9001,
    charging: false,
    maximumCapacity: 99999,
  }),
  [IpcRequest.GetBackupsInfo]: Promise.resolve({
    backups: [
      {
        createdAt: "20-11-15T07:35:01.562Z20",
        size: 99999,
      },
      {
        createdAt: "20-01-30T07:35:01.562Z20",
        size: 1234567,
      },
    ],
  }),
}
