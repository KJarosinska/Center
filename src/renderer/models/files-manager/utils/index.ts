import { FilesManagerState } from "Renderer/models/files-manager/files-manager.interface"

export const prepareDataForStackedBarChart = (data: FilesManagerState) => {
  return data.memoryData.map(el => {
    const { filesType, filesAmount, occupiedMemory, ...rest } = el
    return {
      value: occupiedMemory,
      ...rest,
    }
  })
}
