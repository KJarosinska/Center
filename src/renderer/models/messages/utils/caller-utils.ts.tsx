import { Author } from "Renderer/models/messages/messages.interface"

const removeDecoratorsFromPhoneNumber = (string = ""): string => {
  return string.split(" ").join("").replace("+", "")
}

export interface CallerSearchParams extends Record<string, string> {
  id: string
  phoneNumber: string
}

export const isCallerMatchingSearchParams = (
  caller: Author,
  params: CallerSearchParams
): boolean => {
  return (
    params.id === caller.id &&
    removeDecoratorsFromPhoneNumber(caller.primaryPhoneNumber) ===
      removeDecoratorsFromPhoneNumber(params.phoneNumber)
  )
}
