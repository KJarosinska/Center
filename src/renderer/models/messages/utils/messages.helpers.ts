import { Contact } from "Renderer/models/phone/phone.typings"
import { Topic } from "Renderer/models/messages/messages.interface"

export type ContactsCollection = Record<string, Contact>
export type Getter = (
  id: string,
  collection: ContactsCollection
) => Contact | undefined

export const getContactsAsMap = (contacts: Contact[]) => {
  return contacts.reduce((acc: ContactsCollection, item: Contact) => {
    return {
      ...acc,
      [item.id]: item,
    }
  }, {})
}

export const getContactDetails = (
  id: string,
  collection: ContactsCollection
): Contact | undefined => {
  if (id in collection) {
    return collection[id]
  }

  return undefined
}

export const expandTopic = (
  topic: Topic,
  collection: ContactsCollection,
  getter: Getter
) => {
  const { messages } = topic

  return {
    ...topic,
    messages: messages.map((msg) => {
      const author = getter(msg.author.id, collection)

      if (author) {
        return {
          ...msg,
          author,
        }
      }

      return msg
    }),
  }
}

export const createFullMessagesCollection = (state: any) => {
  const {
    messages: { topics },
    phone: { contacts: baseContacts },
  } = state

  const contacts = getContactsAsMap(baseContacts)

  return topics.map((topic: Topic) => {
    const { id } = topic.caller
    const caller = getContactDetails(id, contacts)

    if (caller) {
      return {
        ...expandTopic(topic, contacts, getContactDetails),
        caller,
      }
    }

    return topic
  })
}
