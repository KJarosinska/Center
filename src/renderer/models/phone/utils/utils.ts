import Faker from "faker"

// TODO: remove before production
export const generateFakeData = (numberOfContacts: number) => {
  return Array(numberOfContacts)
    .fill(0)
    .map(_ => ({
      id: Faker.random.uuid(),
      firstName: Faker.name.firstName(),
      lastName: Faker.name.lastName(),
      phoneNumber: Faker.phone.phoneNumber(),
      favourite: Faker.random.boolean(),
    }))
}

export const generateSortedStructure = (fakeState: any) => {
  const alphabet = new Array(26)
    .fill(1)
    .map((_, i) => String.fromCharCode(65 + i))

  const sortedContactList = fakeState.sort((a: any, b: any) =>
    a.firstName.localeCompare(b.firstName)
  )

  const generateFakeStructure = () => {
    const fakeStructure = []

    for (const letter of alphabet) {
      fakeStructure.push({
        category: letter,
        contacts: [],
      })
    }

    fakeStructure.unshift({
      category: "Favs",
      contacts: [],
    })

    return fakeStructure
  }

  const placeContactsInStructure = () => {
    const fakeStructure: any = generateFakeStructure()
    fakeStructure.forEach((fakeContact: any) => {
      const contactLetter = fakeContact.category
      const contactList: string[] = fakeContact.contacts
      sortedContactList.forEach((sortedContact: any) => {
        if (
          sortedContact.firstName.charAt(0) === contactLetter &&
          !sortedContact.favourite
        ) {
          contactList.push(sortedContact)
        }
      })
    })

    sortedContactList.forEach((contacts: any) => {
      if (contacts.favourite) {
        fakeStructure[0].contacts.push(contacts)
      }
    })

    return fakeStructure
  }

  return placeContactsInStructure()
}

export const sanitizeContacts = (structure: any) => {
  return structure.filter((el: any) => el.contacts.length > 0)
}

export const filterContacts = (contacts: any, substring: string) => {
  return contacts.map((contactsByLetter: any) => ({
    ...contactsByLetter,
    contacts: contactsByLetter.contacts.filter((contact: any) => {
      const filterableFields = Object.values(contact).filter(value => {
        return typeof value === "string"
      })
      return filterableFields.some((value: any) =>
        value.toLowerCase().includes(substring.toLowerCase())
      )
    }),
  }))
}
