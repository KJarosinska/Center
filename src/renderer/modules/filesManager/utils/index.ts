import {
  FilesManagerData,
  FilesManagerState,
} from "Renderer/models/files-manager/files-manager.interface"

export const memoryDataWithConvertedBytes = (data: FilesManagerState) => {
  return data.memoryData.map((element: FilesManagerData) => ({
    ...element,
    occupiedMemory: element.occupiedMemory / Math.pow(1024, 3),
  }))
}
