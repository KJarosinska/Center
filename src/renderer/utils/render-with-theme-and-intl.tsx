import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import { LANGUAGE } from "Renderer/constants/languages"
import localeEn from "Renderer/locales/main/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult<Q>
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q> | Omit<RenderOptions, "queries">
) {
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={LANGUAGE.default}
        locale={LANGUAGE.default}
        messages={localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}

export function renderWithTheme(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithTheme<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult<Q>
export function renderWithTheme<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q> | Omit<RenderOptions, "queries">
) {
  const originalConsoleError = console.error
  console.error = (...args: any) => {
    if (
      !(
        args[0].startsWith("[React Intl] Cannot format message:") ||
        args[0].startsWith("[React Intl] Missing message:")
      )
    ) {
      originalConsoleError.call(console, ...args)
    }
  }
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider locale={LANGUAGE.default}>{ui}</IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
