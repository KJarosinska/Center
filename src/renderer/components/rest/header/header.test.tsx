import "@testing-library/jest-dom/extend-expect"
import React from "react"
import { MemoryRouter } from "react-router-dom"
import Header from "Renderer/components/rest/header/header.component"
import Tabs from "Renderer/components/rest/header/tabs.component"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"

test("matches snapshot without tabs", () => {
  const currentLocation = "/overview"
  const { container } = renderWithThemeAndIntl(
    <MemoryRouter initialEntries={[currentLocation]}>
      <Header middleComponent={<Tabs currentLocation={currentLocation} />} />
    </MemoryRouter>
  )
  const header = container.firstChild
  expect(header).toMatchSnapshot()
})

test("matches snapshot with tabs", () => {
  const currentLocation = "/phone"
  const { container } = renderWithThemeAndIntl(
    <MemoryRouter initialEntries={[currentLocation]}>
      <Header middleComponent={<Tabs currentLocation={currentLocation} />} />
    </MemoryRouter>
  )
  const header = container.firstChild
  expect(header).toMatchSnapshot()
})

test("header should render correct pathname", () => {
  const { getByTestId } = renderWithThemeAndIntl(
    <MemoryRouter initialEntries={["/overview"]}>
      <Header />
    </MemoryRouter>
  )
  expect(getByTestId("location")).toHaveTextContent("Overview")
})
