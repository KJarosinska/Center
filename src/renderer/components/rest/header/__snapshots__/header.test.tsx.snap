// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with tabs 1`] = `
.c1 {
  margin: 0;
  font-size: 1.8rem;
  line-height: 1.1;
  font-weight: 700;
}

.c8 {
  margin: 0;
  font-size: 1.6rem;
  line-height: 1.05;
}

.c0 {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  background-color: #ffffff;
  border-bottom: 0.1rem solid #e6e8eb;
}

.c2 {
  margin: 2.4rem 0 1.5rem 4rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2rem;
  width: 2rem;
  position: relative;
}

.c6 svg {
  width: 100%;
  height: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: 100%;
  padding: 0.8rem;
  border: none;
  position: relative;
  border-radius: 0.2remrem;
  color: #6a6a6a;
  font-weight: 400;
  width: 100%;
}

.c4:after {
  content: "";
  position: absolute;
  display: block;
  bottom: 0;
  left: 0;
  width: 0;
  height: 0.2rem;
  background-color: #000000;
  -webkit-transition: width 500ms ease;
  transition: width 500ms ease;
}

.c4.active {
  color: #000000;
}

.c4.active:after {
  width: 100%;
}

.c4.active svg {
  opacity: 1;
}

.c4 svg {
  opacity: 0.75;
}

.c7 {
  margin: 0 0.8rem 0 0;
}

.c9 {
  white-space: nowrap;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5:not(:last-child) {
  margin-right: 4rem;
}

<div
  class="c0"
>
  <h3
    class="c1 c2"
    data-testid="location"
  >
    Phone
  </h3>
  <div
    class="c3"
  >
    <a
      aria-current="page"
      class="c4 c5 active"
      displaystyle="9"
      href="/phone"
      size="2"
    >
      <div
        class="c6 c7"
        data-testid="icon-Calls"
      >
        <test-file-stub
          image="test-file-stub"
        />
      </div>
      <p
        class="c8 c9"
      >
        Calls
      </p>
    </a>
    <a
      class="c4 c5"
      displaystyle="9"
      href="/phone/dial"
      size="2"
    >
      <div
        class="c6 c7"
        data-testid="icon-Dial"
      >
        <test-file-stub
          image="test-file-stub"
        />
      </div>
      <p
        class="c8 c9"
      >
        Dial
      </p>
    </a>
  </div>
</div>
`;

exports[`matches snapshot without tabs 1`] = `
.c1 {
  margin: 0;
  font-size: 1.8rem;
  line-height: 1.1;
  font-weight: 700;
}

.c0 {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  background-color: #ffffff;
  border-bottom: 0.1rem solid #e6e8eb;
}

.c2 {
  margin: 2.4rem 0 1.5rem 4rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
>
  <h3
    class="c1 c2"
    data-testid="location"
  >
    Overview
  </h3>
  <div
    class="c3"
  />
</div>
`;
